### DISCUSSION TITLE
DatePicker Focus and scroll top problem

### DISCUSSION THREAD

**Discussion by fsachet** [microsoft/fluentui#29883] [Q&A] [answered]
Hello, 
I open a new discussion because I can't foud an answer of my problem. I acutally used datePicker (office-ui-fabric-react version 7.204.0) in SPFX project.

When I open the calendar using the icon and I select a date, the scoll of the page go on top. If i checked the document.activeElement is the main role element. If I clicked on the combobox element of the DatePicker, I don't have this problem.

Has anyone had an idea of this problem ?

Thanks in advanced for your help.

**Comment by arasubala** [comment]
@fsachet - did you manage to resolve this? I have the same problem with DatePicker in SPFx solution.

**Reply by fsachet** [reply]
I currently don't have found a solution to resolve this problem. I updated to use the last version of the library but I have the same behavior.

I checked the code used by the click event on the icon and  combobox element and I remarked that is not the same.

I suppose that an update should be applied to the library for resolved this problem but I didn't have time to investigate further.

**Reply by HiltonGiesenow** [reply]
I'm also seeing this in several places in the same solution. SPFx 1.18.2 with @fluentui/react 8.106.4 ...

**Reply by kasperbolarsen** [reply]
Same here for the PnP Modern Search project, https://github.com/microsoft-search/pnp-modern-search/issues/4321

**Comment by Macro21** [comment, marked as answer]
Through testing, I have discovered that the error only occurs when the user clicks on the calendar icon. 
Therefore, I have found an alternative solution that has worked for me. This solution involves adding a 'pointer-events: none' CSS property to the date picker icon. This way, I have managed to solve the problem.

![image](https://github.com/microsoft/fluentui/assets/24574303/cc864c8a-80af-405f-bc9b-d231875b169a)
![image](https://github.com/microsoft/fluentui/assets/24574303/51d31afd-0cbe-45a5-81f9-d3346431ca58)

**Reply by fsachet** [reply]
Thanks a lot for your message and your alternative solution.
I also confirm that is resolving my problem if i add the css property

**Reply by Kennybestman** [reply]
Thank you very much for your knowledge share. I also attest that this resolved my issue when I added the css property as suggested

**Reply by alexbbell** [reply]
Thank you! There's an alternative way to add css. You can write them in the component styles.

<DatePicker
	firstDayOfWeek={DayOfWeek.Monday} // Monday as the first day of the week
	styles={{
		root: {
		fontFamily: 'inherit',
		color: 'inherit',
		width: '100%', // Force full width to prevent layout shifts
		},
		**icon: {
			pointerEvents: 'none'
		},**
	statusMessage: {
		fontWeight: 'normal'
	} }}  .... />
